### PR TITLE
Feature/formViewController

### PR DESCRIPTION
**Pull request by 3lvis**
- [x] Added `FORMViewController`, this should help to reduce boilerplate.
- [x] Updated the Basic-ObjC example to use this.
- [x] Made an issue to update the other demos => https://github.com/hyperoslo/Form/issues/393

### CODE REVIEW

**Comment by 3lvis** [comment]
https://github.com/hyperoslo/Form/issues/393

**Comment by 3lvis** [comment]
Fixes => #255

File: Demos/Basic-ObjC/Basic-ObjC/Controllers/HYPSampleCollectionViewController.m [hunk 1]
```diff
@@ -16,57 +16,58 @@
...
-- (void)dealloc {
-    [self hyp_removeKeyboardToolbarObservers];
-}
-
 #pragma mark - Initialization
 
 - (instancetype)initWithJSON:(NSArray *)JSON andInitialValues:(NSDictionary *)initialValues {
-    FORMLayout *layout = [[FORMLayout alloc] init];
-    self = [super initWithCollectionViewLayout:layout];
+    self = [super initWithJSON:JSON andInitialValues:initialValues];
```
Comment (zenangst) [inline Demos/Basic-ObjC/Basic-ObjC/Controllers/HYPSampleCollectionViewController.m]: Code style `:`

File: Source/FORMViewController.m [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+#pragma mark - Initialization
+
+- (instancetype)initWithJSON:(id)JSON
+            andInitialValues:(NSDictionary *)initialValues {
+    FORMLayout *layout = [[FORMLayout alloc] init];
+    self = [super initWithCollectionViewLayout:layout];
+    if (!self) return nil;
+
+    _JSON = JSON;
+    self.layout = layout;
```
Comment (zenangst) [inline Source/FORMViewController.m]: Should this be `_layout` instead of `self.layout`?

Comment (zenangst) [inline Source/FORMViewController.m]: Or maybe that only goes for non-inherited properties.

**Comment by zenangst** [comment]
Looks good, like the idea behind `FORMViewController`.

File: Source/FORMViewController.m (hunk 2 above)
Comment (3lvis) [inline Source/FORMViewController.m]: That would work, I'll change it

**Comment by 3lvis** [comment]
Done!

**Comment by 3lvis** [comment]
@zenangst Fixed!

**Comment by zenangst** [comment]
@3lvis :cake:

**Pull request merged**